src/browser/transports/unix_domain_socket.ts

```diff
@@ -0,0 +1,84 @@
+import { app } from 'electron';
+import { unlink } from 'fs';
+import { exec } from 'child_process';
+
+// we need to use require here because unix-dgram is an optional dependency
+// and is only available on unix systems. import does a static check and
+// will not compile if the module isn't present at time of compilation
+const unixDgram = process.platform === 'win32' ? {} : require('unix-dgram');
+
+import BaseTransport from './base';
+
+type FileDescriptor = string;
+
+interface Socket {
+    close: () => void;
+    bind: (name: string) => void;
+    on: (event: string, callback?: (...args: any[]) => void) => void;
+    send: (buffer: Buffer, offset: number, length: number, path: FileDescriptor) => void;
+}
+
+class UnixDomainSocket extends BaseTransport {
+    private fileDescriptorPrefix: string;
+    private serverName: FileDescriptor;
+    private server: Socket;
+
+    constructor(fileDescriptorPrefix: string) {
+        super();
+
+        this.fileDescriptorPrefix = fileDescriptorPrefix;
+        this.serverName = fileDescriptorPrefix + Date.now();
+        this.server = unixDgram.createSocket('unix_dgram', (buffer: Buffer) => {
+            this.eventEmitter.emit('message', null, buffer);
+        });
+        this.server.bind(this.serverName);
+
+        app.on('will-quit', () => {
+            this.cleanUpServer();
+        });
+
+        process.on('SIGINT', () => {
+            this.cleanUpServer();
+        });
+    }
+
+    public publish(data: any): boolean {
+        const message = new Buffer(JSON.stringify(data));
+        this.getFileDescriptors().then((fds: FileDescriptor[]) => {
+            fds
+                .filter((fd: FileDescriptor) => fd !== this.serverName)
+                .forEach((fd: FileDescriptor) => {
+                    const client: Socket = unixDgram.createSocket('unix_dgram');
+                    client.send(message, 0, message.length, fd);
+                    client.close();
+                });
+        });
+        return true;
+    }
+
+    private cleanUpServer() {
+        this.server.close();
+        unlink(this.serverName);
+    }
+
+    private getFileDescriptors(): Promise<FileDescriptor[]> {
+        return new Promise<FileDescriptor[]>((resolve) => {
+            exec(`lsof -U | grep ${this.fileDescriptorPrefix}`, (error, stdout) => {
```

Comment:
We get all open file descriptors that contain the prefix by using command line tools and then parsing the output.